File: python/balance_parens.py
def balance_parens(str):
    split_str = list(str)
    stack = []

    for idx, char in enumerate(split_str) :
        if char == ")" :
            if stack:
                stack.pop()
            else:
                split_str[idx] = ''
        elif char == "(" :
            stack.append(idx)
    
    for i in stack :
        split_str[i] = ''

    return ''.join(split_str)

File: js/balanceParens.js
balanceParens = (str) => {
    let strSplit = str.split('');
    let stack = [];

    for (let i = 0; i < strSplit.length; i++) {
        if (strSplit[i] === ')') {
            if (stack.length !== 0) {
                stack.pop()
            } else {
                strSplit[i] = ''
            }
        } else if (strSplit[i] === '(') {
            stack.push(i)
        }
    } 

    for (let i = 0; i < stack.length; i++) {
        strSplit[stack[i]] = ''
    }
    return strSplit.join('')

}

module.exports = { balanceParens }